src/query.c

```diff
@@ -289,6 +289,20 @@ static void setFilterNode(QueryAST *q, QueryNode *n) {
     // we usually want the numeric range as the "leader" iterator.
     q->root->children = array_ensure_prepend(q->root->children, &n, 1, QueryNode *);
     q->numTokens++;
+  // vector node should always be in the root, so we have a special case here.
```

Comment:
is this the right place to have this modification?
`setFilterNode` is called also when `options->numeric` and `options->geo` are true. How this change affects those calls if any?

Reply:
I think so (also consulted with @ashtul). I will add tests for the legacy numeric and geo global filters, as the fix holds for these cases as well.